src/WSSecurity/Entry/Encryption.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+namespace Soap\Psr18WsseMiddleware\WSSecurity\Entry;
+
+use RobRichards\WsePhp\WSSESoap;
+use RobRichards\XMLSecLibs\XMLSecurityKey;
+use Soap\Psr18WsseMiddleware\WSSecurity\KeyStore\KeyInterface;
+use Soap\Psr18WsseMiddleware\WSSecurity\SignatureAlgorithm;
+use VeeWee\Xml\Dom\Document;
+
+final class Encryption implements WsseEntry
+{
+    private KeyInterface $key;
+    private SignatureAlgorithm $symmetricEncodingAlgorithm = SignatureAlgorithm::AES256_CBC; // TODO Move out to SymmetricEncodingAlgorithm
+    private SignatureAlgorithm $keyEncryptionAlgorithm = SignatureAlgorithm::RSA_OAEP_MGF1P; // TODO Move out to KeyEncryptionAlgorithm
+
+    // TODO : SHould the keyinfo be applied by a KeyIdentifier instead?
+    private bool $serverCertificateHasSubjectKeyIdentifier = true;
+
+    public function __construct(KeyInterface $key)
+    {
+        $this->key = $key;
+    }
+
+    public function __invoke(Document $envelope, WSSESoap $wsse): void
+    {
+        $key = new XMLSecurityKey($this->symmetricEncodingAlgorithm->value);
+        $key->generateSessionKey();
+
+        $siteKey = new XMLSecurityKey($this->keyEncryptionAlgorithm->value, ['type' => 'public']);
+        $siteKey->passphrase = $this->key->passphrase();
+        $siteKey->loadKey($this->key->contents(), false, $this->key->isCertificate());
+
+        $wsse->encryptSoapDoc($siteKey, $key, [
+            'KeyInfo' => [
```

Comment:
similar logic is available for signing. Maybe better to move it out to a separate key identifier instead;